### PR TITLE
Add pedersen hash builtin

### LINKED ISSUE #6: feat: implement pedersen hash builtin
### Description
A cell is constrained by values in two other cells if these memory cells are initialized with the Pedersen hash builtin. 
Linked to the [issue](https://github.com/keep-starknet-strange/cairo-zig/issues/22) to implement the primitive first 

### Acceptance Criteria
Implementation and passing tests. 

### Resources
See [test](https://github.com/NethermindEth/cairo-vm-go/blob/main/pkg/vm/builtins/pedersen_test.go) to understand the logic. 

See [Rust](https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/vm/runners/builtin_runner/hash.rs) and [Go](https://github.com/NethermindEth/cairo-vm-go/blob/main/pkg/vm/builtins/pedersen.go) implementations

### PR DESCRIPTION
**Pull request by edisontim**
Closes #6

### CODE REVIEW

File: src/vm/builtins/builtin_runner/builtin_runner.zig [hunk 1]
```diff
@@ -122,7 +122,10 @@ pub const BuiltinRunner = union(enum) {
         return switch (self.*) {
             .Bitwise => |bitwise| try bitwise.deduceMemoryCell(address, memory),
             .EcOp => |ec| ec.deduceMemoryCell(address, memory),
-            .Hash => |hash| hash.deduceMemoryCell(address, memory),
+            .Hash => |hash| {
```
[the remark this answers is not in the document]
Comment (edisontim) [reply src/vm/builtins/builtin_runner/builtin_runner.zig:125]: Indeed, didn't notice this

[the remark this answers is not in the document]
Comment (tcoratger) [reply src/vm/builtins/builtin_runner/builtin_runner.zig]: You are almost there, now you don't need to declare a mut var, you can directly use

```zig
.Hash => |*hash|  hash.deduceMemoryCell(address, memory),
```

[the remark this answers is not in the document]
Comment (edisontim) [reply src/vm/builtins/builtin_runner/builtin_runner.zig]: Of course yeah, my bad. I'm a bit distracted

[the remark this answers is not in the document]
Comment (edisontim) [reply src/vm/builtins/builtin_runner/builtin_runner.zig]: Should be good now

**Review by lana-shanghai — APPROVED** [review]
lgtm

**Pull request merged**